src/main/java/com/epam/ta/reportportal/binary/CreateLogAttachmentService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.ta.reportportal.binary;
+
+import com.epam.ta.reportportal.dao.LogRepository;
+import com.epam.ta.reportportal.entity.attachment.Attachment;
+import com.epam.ta.reportportal.entity.log.Log;
+import com.epam.ta.reportportal.exception.ReportPortalException;
+import com.epam.ta.reportportal.ws.model.ErrorType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+@Service
+@Transactional
+public class CreateLogAttachmentService {
```

Comment:
do we really need separate service for that?

Reply:
Yes, it should execute in one separate transaction

Reply:
would placing this stuff into separate method (within existing handler) and marking it with `@Transactional(requiresNew)` solve the issue?

Reply:
No. Self-invocation will not start transaction even if it market with annotation